daemon/container_unit_test.go

```diff
@@ -195,3 +196,31 @@ func TestValidContainerNames(t *testing.T) {
 		}
 	}
 }
+
+func TestFindSignificantParent(t *testing.T) {
+	paths := []string{
+		"/etc/foo/bar/baz/qux",
+		"/etc/foo/qux",
+		"/etc/foo",
+		"/etc/foo/bar/baz",
+		"/etc/bar",
+		"/etc/foo/bar",
+	}
+
+	if b := findSignificantParent(paths); b != "/etc" {
+		t.Fatalf("Expected /etc, was %s\n", b)
+	}
+
+	paths = []string{"/etc/foo"}
+	if b := findSignificantParent(paths); b != "/etc" {
+		t.Fatalf("Expected /etc, was %s\n", b)
+	}
+
+	paths = []string{
+		"/etc/foo",
+		"/etc/bar/baz",
+	}
+	if b := findSignificantParent(paths); b != "/etc" {
+		t.Fatalf("Expected /etc, was %s\n", b)
+	}
+}
```

Comment:
Let's add a test for things like:
```
/etc/foo/bar
/tmp/boogie/nights
```
to make sure it comes up with `/`.

And:
```
/tmp/dir1/dir2/dir3/dir4/file
/tmp/dir1/dir3/dir4/dir5/file
```
results in `/tmp/dir1` to make sure we walk up more than one level when needed.